config/initializers/slow_ride.rb

```diff
@@ -0,0 +1,37 @@
+require "redis"
+require "slow_ride"
+
+SlowRide.enable_redis do
+  Redis.new(url: ENV["REDIS_SLOW_RIDE_URL"] || ENV["REDIS_URL"])
+end
+
+# 20% Failure rate
+# 100 checks
+# Reset after it hasn't been checked in 5 minutes
+MY_FEATURE = SlowRide::Redis.new(:my_feature, failure_threshold: 0.2, minimum_checks: 100, max_duration: 5.minutes) do
+  Flipper.disable :my_feature
+  Rails.logger.warn "Flipper my_feature disabled"
+end
+
+# Example usage of SlowRide with Flipper
+#
+# To test:
+#   # Run 80x doing definitely very normal things
+#   80.times { check_slow_ride { 123 } }
+#   Flipper.enabled?(:my_feature) # => true
+#
+#   # Run another 25x where it errors out
+#   19.times { check_slow_ride { raise "hell" } rescue nil }
+#   Flipper.enabled?(:my_feature) # => true, we haven't reached the failure threshold or the min checks yet
+#
+#   # One more error
+#   check_slow_ride { raise "hell" } rescue nil
+#   Flipper.enabled?(:my_feature) # => false, automatically disabled
+def check_slow_ride(&block)
+  if FeatureFlag.enabled?(:my_feature)
+    # Feature flag is on
+    MY_FEATURE.check(&block)
```

Comment:
I like it! what's the chance of this detecting false negatives? and what would be the process of restarting slow-ride on the same :feature?  I would imagine a slow-ride experiment is tied to the :name

Reply:
@maestromac

> what's the chance of this detecting false negatives?

Right now it only fails a check on exception. I was also considering other failure modes that don't necessarily result in an exception but I wasn't sure how to represent them. Maybe something like this?

```ruby
MY_FEATURE.check do |check|
  result = do_some_work

  if result == :not_what_we_were_hoping_for
    check.fail!
  end

  result
end
```

> what would be the process of restarting slow-ride on the same :feature?

The check deletes itself after invoking the block, so if we follow the convention of disabling the feature flag on failure, restarting the check is simply a matter of re-enabling the feature flag.

If we don't, the block will continue to be invoked every time it meets the requirements. So, for example, the block could be invoked every 1000 requests (`minimum_checks`) as long as the failure rate remains above 20% (`failure_threshold`). I wouldn't want to use this as a common pattern, but I'm sure scenarios exist where this is useful.

>  I would imagine a slow-ride experiment is tied to the :name

Yes, exactly. It's represented in Redis as `slow-ride:#{name}:checked` and `...:failed`.